### PR TITLE
Song model

### PR DESCRIPTION
**Pull request by JosephFarina**

### CODE REVIEW

File: app/Global.java [hunk 1]
```diff
@@ -45,16 +46,21 @@ public void onStart(Application app) {
...
+            ussr.save();
+            Song dP = new Song("Dear Prudence", ((3 * 60) + 56L), 2, Album.find.where().eq("name", "The White Album").findUnique() );
+            dP.save();
+            Song warmGun = new Song("Happiness Is a Warm Gun", ((2 * 60) + 43L), 8, Album.find.where().eq("name", "The White Album").findUnique() );
+            warmGun.save();
+            Song goodTimes = new Song("Good Times Bad Times", ((2 * 60) + 46L), 1, Album.find.where().eq("name", "Led Zeppelin 1").findUnique() );
+            goodTimes.save();
         }
 
+//        public Song(String name, Long length, Album album)
```
Comment (nickfarina) [inline app/Global.java]: :boom:

File: app/controllers/AlbumController.java [hunk 2]
```diff
@@ -16,4 +16,9 @@ public Result index() {
         return ok(albumIndex.render(albumList));
     }
 
+    public Result show(Long id) {
+
```
Comment (nickfarina) [inline app/controllers/AlbumController.java]: :boom:

File: app/controllers/ArtistController.java [hunk 3]
```diff
@@ -14,14 +14,12 @@ public Result index() {
 
         List<Artist> artistList = Artist.find.all();
         return ok(artists.render(artistList));
-
     }
 
     public Result show(Long id) {
```
Comment (nickfarina) [inline app/controllers/ArtistController.java]: :boom:

Comment (JosephFarina) [inline app/controllers/ArtistController.java]: what in goodness name does that mean lol

File: app/models/Album.java [hunk 4]
```diff
@@ -20,6 +20,9 @@
     @ManyToOne(optional = false)
     public Artist artist;
 
+    @OneToMany
+    public List<Song> songs = new ArrayList<Song>();
```
Comment (nickfarina) [inline app/models/Album.java]: get rid of this, make `getSongs` method

File: app/models/Album.java [hunk 5]
```diff
@@ -29,10 +32,10 @@ public Album(String name, Boolean explicit, Artist owner){
 
     public static Model.Finder<Long, Album> find = new Model.Finder<Long, Album>(Long.class, Album.class);
 
-    public static List<Album> findAlbumsByArtist(Artist a) {
-        return find.fetch("artist")
-                    .where()
-                    .eq("artist.name", a.name)
+    public List<Song> getSongs(Album album) {
```
Comment (nickfarina) [inline app/models/Album.java]: why are you passing an album into this?

File: app/models/Artist.java [hunk 6]
```diff
@@ -28,4 +28,11 @@ public Artist(String name, String genre, String bio) {
     public static Artist uniqueArtist(String artistName) {
         return Artist.find.where().eq("name", artistName).findUnique();
     }
+
+    public List<Album> getAlbums(Artist artist) {
```
Comment (nickfarina) [inline app/models/Artist.java]: why are you passing artist in

Comment (JosephFarina) [inline app/models/Artist.java]: So you can go Artist.getAlbums() and get a List of their albums from that?

File: app/models/Album.java (hunk 5 above)
Comment (JosephFarina) [inline app/models/Album.java]: so you can go Album.getSongs and get all the songs that are associated with the album.

File: app/controllers/ArtistController.java (hunk 3 above)
Comment (nickfarina) [inline app/controllers/ArtistController.java]: get rid of this line. generally look like this - 
```java
public Result show(Long id) {
  Album uniqueAlbum = Album.find.byId(id);

  return ok(albumPage.render(uniqueAlbum));
}
```
or
```java
public Result show(Long id) {
  Album uniqueAlbum = Album.find.byId(id);
  return ok(albumPage.render(uniqueAlbum));
}
```

File: app/models/Album.java (hunk 5 above)
Comment (nickfarina) [inline app/models/Album.java]: but you're using it like this in the template `album.getSongs(album)`

looks redundant doesn't it? 

if it's a non static method (aka instance method) then you can only call it on the instance of a variable, and if you're calling it on an instance of a variable, then you don't need to pass the same variable in - you have access to it inside of the method using `this`

Comment (JosephFarina) [inline app/models/Album.java]: ohhhh. Yes I thought that as i was writing it in the view. 

So keep the method but make it just public List<Song> getSongs()? 
and where it says album.whatever would be this.whatever?

Comment (nickfarina) [inline app/models/Album.java]: ya, only inside of the method body tho. so - 

```java
return Song.find
  .where()
  .eq("album", this)
  .findList();
```

Comment (JosephFarina) [inline app/models/Album.java]: these r the things to know. niceee.

**Pull request closed without merging**